src/queueManager.js

```diff
@@ -186,6 +186,12 @@ export default class QueueManager {
           if (response.KVAPID) {
             Utils.log.debug(`kaios vapid recieved: ${response.KVAPID}`);
             Device.setVAPID(response.KVAPID);
+            for(i = 0; i < events.length; i++){
```

Comment:
Should be moved outside `if(response.KVAPID)` block.

Reply:
@royson-ct if I move it outside the `if(response.KVAPID)` block, then for every App Launched event the `Device.setVAPIDState(true);` will be set always beating the point of fixing this bug

Reply:
@darshanclevertap initially the VAPIDState would be falsy (`getVAPIDState()`). Once the response for "App Launched" is received it would be set as true. Doesn't matter how many consecutive calls are made for "App Launched".
When inside the `if` block. For accounts without `KVAPID` set on dashboard `getVAPIDState()` would always be falsy causing setTimeout to run infinitely.

Reply:
@royson-ct Got it. Will make the changes 👍🏾